pkg/controller/node/taint_controller_test.go

```diff
@@ -464,12 +502,57 @@ func TestUpdateNode(t *testing.T) {
 			expectDelete:    false,
 			additionalSleep: 1500 * time.Millisecond,
 		},
+		{
+			description: "Pod with multiple tolerations are victed when first one runs out",
+			pods: []v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "default",
+						Name:      "pod1",
+						Annotations: map[string]string{
+							"scheduler.alpha.kubernetes.io/tolerations": `
+  [
+    {
+      "key": "testTaint1",
+      "value": "test1",
+      "effect": "` + string(v1.TaintEffectNoExecute) + `",
+      "tolerationSeconds": ` + fmt.Sprintf("%v", 1) + `
+    },
+    {
+      "key": "testTaint2",
+      "value": "test2",
+      "effect": "` + string(v1.TaintEffectNoExecute) + `",
+      "tolerationSeconds": ` + fmt.Sprintf("%v", 100) + `
+    }
+  ]
+  `,
+						},
+					},
+					Spec: v1.PodSpec{
+						NodeName: "node1",
+					},
+					Status: v1.PodStatus{
+						Conditions: []v1.PodCondition{
+							{
+								Type:   v1.PodReady,
+								Status: v1.ConditionTrue,
+							},
+						},
+					},
+				},
+			},
+			oldNode:         testutil.NewNode("node1"),
+			newNode:         addTaintsToNode(testutil.NewNode("node1"), "testTaint1", "taint1", []int{1, 2}),
+			expectDelete:    true,
```

Comment:
I don't understand. We sleep for 1.7s here, so there's no chance of noticing eviction after 100s.